src/System.Management.Automation/engine/parser/SymbolResolver.cs

```diff
@@ -730,11 +730,11 @@ internal static string GetModuleQualifiedName(string namespaceName, string typeN
         }
     }
 
-    internal class SymbolResolvePostActionVisitor : DefaultCustomAstVisitor2
+    internal class SymbolResolvePostActionVisitor : ICustomAstVisitor2
```

Comment:
If we're keeping both `DefaultCustomAstVisitor2` and `ICustomAstVisitor2`, is it worth us changing this? Perhaps the answer is yes in terms of us "endorsing" the correct API. To weigh another consideration, I did have a quick look through https://github.com/dotnet/runtime/issues/7291, but I can't imagine it would be much of an impact to us

Reply:
> I did have a quick look through [dotnet/runtime#7291](https://github.com/dotnet/runtime/issues/7291), but I can't imagine it would be much of an impact to us

I think the cost is already incurred here since the caller (`Ast.Visit`) was already referencing the interface.  I could be reading that issue wrong though.

Reply:
Yeah, I think you're right

Reply:
I reverted the changes to `SymbolResolvePostActionVisitor`. It's a good example showing that code derives from `DefaultCustomAstVisitor2` would possibly need no change even there are new AST types added.